doc/api/api.yaml

```
@@ -2854,7 +2854,192 @@ paths:
+        - $ref: "#/components/parameters/XApiPrefixHeader"
+      responses:
+        '302':
+          description: Response
+          headers:
+            Location:
+              schema:
+                type: string
+                description: login page or error page url
+  /auth/logout:
+    get:
+      summary: Redirect to logout url
+      operationId: handler-logout-get
+      tags:
+        - main
+      parameters:
+        - $ref: "#/components/parameters/XForwardedProtoHeader"
+        - $ref: "#/components/parameters/XForwardedHostHeader"
+        - $ref: "#/components/parameters/XApiPrefixHeader"
+        - $ref: "#/components/parameters/SessionCookie"
+      responses:
+        '302':
+          description: Response
+          headers:
+            Location:
+              schema:
+                type: string
+                description: logout page or error page url
+  /auth/login/callback:
+    get:
+      summary: Login callback
+      operationId: handler-login-callback-get
+      tags:
+        - main
+      parameters:
+        - $ref: "#/components/parameters/redirectPath"
+        - $ref: "#/components/parameters/state"
+        - $ref: "#/components/parameters/code"
+        - $ref: "#/components/parameters/UserAgentHeader"
+        - $ref: "#/components/parameters/XForwardedProtoHeader"
+        - $ref: "#/components/parameters/XForwardedHostHeader"
+        - $ref: "#/components/parameters/XApiPrefixHeader"
+      responses:
+        '302':
+          description: Response
+          headers:
+            Location:
+              schema:
+                type: string
+                description: login page or error page url
+            Set-Cookie:
+              schema:
+                type: string
+                example: session=abcde12345; Path=/; HttpOnly; SameSite=Lax; Secure
+  /auth/logout/callback:
+    get:
+      summary: Logout callback
+      operationId: handler-logout-callback-get
+      tags:
+        - main
+      parameters:
+        - $ref: "#/components/parameters/XForwardedProtoHeader"
+        - $ref: "#/components/parameters/XForwardedHostHeader"
+        - $ref: "#/components/parameters/XApiPrefixHeader"
+      responses:
+        '302':
+          description: Response
+          headers:
+            Location:
+              schema:
+                type: string
+                description: login page or error page url
+  /auth/token/introspect:
+    get:
+      summary: Introspect token
+      operationId: handler-token-introspect-get
+      tags:
+        - main
+      parameters:
+        - $ref: "#/components/parameters/UserAgentHeader"
+        - $ref: "#/components/parameters/SessionCookie"
+      responses:
+        '200':
+          description: Null response
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
```

Давай явно указывать код. И давай сократим запись:
```suggestion
        '500':
          $ref: "#/components/schemas/Error500"
```

---

Готово - только эта ручка может вернуть либо 200, либо 401, поэтому поменял 500 на 401.

Будем менять это везде во всех ручках? Кстати, мы ведь никак пока не используем responses, описанные в ямле?

---

Конкретно эта ручка да. Но большинство других ручек могут вернуть 500. Нужно описать все коды которые возвращаются нами (а не проксёй или фреймворком, например).
Это нужно для всех ручек несмотря на то что пока не используем. Этот ямл ценен сам по себе как дока, и пока это в основном для целей документирования. Но думаю будут и редкие кейсы когда будем использовать сгенерированные из этого файла ответы во фронте.

---

Готово (везде в текущей ветке).

---

Только надо ещё изучить табличные классы и сервис дополнительно, оттуда тоже могут выбрасываться исключения, а следовательно - будут новые статусы ответов. В процессе.